### PR TITLE
Docs section for SLURM troubleshooting

### LINKED ISSUE #14781: stuck at "Initializing distributed.." when using ddp with multiple gpus
### First check

- [x] I'm sure this is a bug.
- [X] I've added a descriptive title to this bug.
- [X] I've provided clear instructions on how to reproduce the bug.
- [X] I've added a code sample.
- [X] I've provided any other important info that is required.

### Bug description

Dear community,

I'm desperately trying to achieve multi gpu training on our scientific SLURM cluster. It has one gpu (Tesla T4) per node, so specifically I want to achieve multi node multi gpu training. For testing I just use a minimal example from PL documentation (https://pytorch-lightning.readthedocs.io/en/latest/starter/introduction.html) to ensure that the error is not build in my model. 
Even when allocating only 1 gpu the code hangs. When using "dp" instead of "ddp" it runs (though not really faster). 

### How to reproduce the bug

```python
MODELL:
import os
import torch
from torch.utils.data import DataLoader, Dataset
from pytorch_lightning import LightningModule, Trainer


class RandomDataset(Dataset):
    def __init__(self, size, length):
        self.len = length
        self.data = torch.randn(length, size)

    def __getitem__(self, index):
        return self.data[index]

    def __len__(self):
        return self.len


class BoringModel(LightningModule):
    def __init__(self):
        super().__init__()
        self.layer = torch.nn.Linear(32, 2)

    def forward(self, x):
        return self.layer(x)

    def training_step(self, batch, batch_idx):
        loss = self(batch).sum()
        self.log("train_loss", loss)
        return {"loss": loss}

    def validation_step(self, batch, batch_idx):
        loss = self(batch).sum()
        self.log("valid_loss", loss)

    def test_step(self, batch, batch_idx):
        loss = self(batch).sum()
        self.log("test_loss", loss)

    def configure_optimizers(self):
        return torch.optim.SGD(self.layer.parameters(), lr=0.1)


def run():
    train_data = DataLoader(RandomDataset(32, 64), batch_size=2, num_workers=4)
    val_data = DataLoader(RandomDataset(32, 64), batch_size=2, num_workers=4)
    test_data = DataLoader(RandomDataset(32, 64), batch_size=2, num_workers=4)

    model = BoringModel()
    trainer = Trainer(
        default_root_dir=os.getcwd(),
        limit_train_batches=1,
        limit_val_batches=1,
        limit_test_batches=1,
        num_sanity_val_steps=0,
        max_epochs=1,
        enable_model_summary=False,
        accelerator="gpu",
        devices=1,
        strategy="ddp",
        num_nodes=4
    )
    trainer.fit(model, train_dataloaders=train_data, val_dataloaders=val_data)
    trainer.test(model, dataloaders=test_data)


if __name__ == "__main__":
    run()


SLURM SUBMISSION SCRIPT:
#!/bin/bash
#SBATCH -J test
#SBATCH -e test.err
#SBATCH -o test.log
#SBATCH --tasks-per-node 8
#SBATCH --gres gpu:1
#SBATCH --nodes 4

SECONDS=0
python train.py
echo "$SECONDS seconds passed"
```

### Error messages and logs

```
GPU available: True (cuda), used: True
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs
`Trainer(limit_train_batches=1)` was configured so 1 batch per epoch will be used.
`Trainer(limit_val_batches=1)` was configured so 1 batch will be used.
`Trainer(limit_test_batches=1)` was configured so 1 batch will be used.
Initializing distributed: GLOBAL_RANK: 0, MEMBER: 1/4
```

### Important info

```
cuda 11.2
torch 1.12.1
pytorch-lightning 1.7.6
```

### More info

I tried out various things from many threads but still can't even get this minimal example to work :(

This is my first bug report ever, please don't hate and thanks a lot in advance!

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #14781

My proposal is to add a section to the docs that explains the most likely cause for hangs. 
Ideally we would provide a useful error message. The issue with a concrete solution is here https://github.com/Lightning-AI/lightning/issues/10150 and here #14078. However, these still can't eliminate all user error, especially regarding the setup of the SLURM submission script. 

@Felonious-Spellfire for review of the language and structure.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @borda @rohitgr7 @Felonious-Spellfire @akihironitta

### CODE REVIEW

File: docs/source-pytorch/clouds/cluster_advanced.rst [hunk 1]
```diff
@@ -195,11 +195,39 @@ See also the multi-node examples
...
+    Initializing distributed: GLOBAL_RANK: 0, MEMBER: 1/4
+
+
+The most likely reasons and how to fix it:
+
+- You forgot to run the `python train.py` command with `srun`:
+  Please have a look at the SLURM template script above which includes the `srun` at the botton of the script.
+
+- The number of nodes or number of devices per node is configured incorrectly:
+  There are two parametres in the SLURM submission script that determine how many processes will run your training, the `#SBATCH --nodes=X` setting and `#SBATCH --ntasks-per-node=Y` settings.
```
Comment (Borda) [inline docs/source-pytorch/clouds/cluster_advanced.rst:220]:
```suggestion
  There are two parametres in the SLURM submission script that determine how many processes will run your training, the ``#SBATCH --nodes=X`` setting and ``#SBATCH --ntasks-per-node=Y`` settings.
```

**Review by Borda — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by otaj — APPROVED** [review]
